### PR TITLE
[db] Add missing Workspace index (id, deleted)

### PR DESCRIPTION
**Pull request by geropl**
## Related Issue(s)

Fixes #

## Release Notes

```release-note
fix missing index on Workspace (id, deleted)
```

### CODE REVIEW

File: components/gitpod-db/src/typeorm/entity/db-workspace.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { TypeORM } from "../typeorm";
 import { Transformer } from "../transformer";
 
 @Entity()
+@Index("ind_id_deleted", ["id", "deleted"])
```
Comment (easyCZ) [inline components/gitpod-db/src/typeorm/entity/db-workspace.ts:21]:
```suggestion
@Index(IndexWorkspaceIdDeleted1647333804782.TABLE_NAME, IndexWorkspaceIdDeleted1647333804782.FIELDS)
```

File: components/gitpod-db/src/typeorm/migration/1647333804782-IndexWorkspaceIdDeleted.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { indexExists } from "./helper/helper";
+
+export class IndexWorkspaceIdDeleted1647333804782 implements MigrationInterface {
+    static readonly TABLE_NAME = "d_b_workspace";
+    static readonly INDEX_NAME = "ind_id_deleted";
+
```
Comment (easyCZ) [inline components/gitpod-db/src/typeorm/migration/1647333804782-IndexWorkspaceIdDeleted.ts:13]:
```suggestion
static readonly FIELDS = ["id", "deleted"]
```

**Review by easyCZ — CHANGES_REQUESTED** [review]

File: components/gitpod-db/src/typeorm/migration/1647333804782-IndexWorkspaceIdDeleted.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (
+            !(await indexExists(
+                queryRunner,
+                IndexWorkspaceIdDeleted1647333804782.TABLE_NAME,
+                IndexWorkspaceIdDeleted1647333804782.INDEX_NAME,
+            ))
+        ) {
+            await queryRunner.query(
+                `CREATE INDEX ${IndexWorkspaceIdDeleted1647333804782.INDEX_NAME} ON ${IndexWorkspaceIdDeleted1647333804782.TABLE_NAME} (id, deleted)`,
```
Comment (easyCZ) [inline components/gitpod-db/src/typeorm/migration/1647333804782-IndexWorkspaceIdDeleted.ts:23]:
```suggestion
                `CREATE INDEX ${IndexWorkspaceIdDeleted1647333804782.INDEX_NAME} ON ${IndexWorkspaceIdDeleted1647333804782.TABLE_NAME} (IndexWorkspaceIdDeleted1647333804782.FIELDS.join(', )`,
```

**Review by easyCZ — APPROVED** [review]
LGTM, thanks for the changes!

**Comment by geropl** [comment]
> thanks for the changes!

@easyCZ thx for spotting! :eyes:

**Comment by geropl** [comment]
Wait somehow, GitHub is fooling me with the PR state... switching betwen closed (not idea how that happened) and open. Starting fresh now.

**Comment by geropl** [comment]
Switching over to #8821

**Pull request closed without merging**